addons/portal/controllers/portal.py

```diff
@@ -550,6 +556,21 @@ def _create_or_update_address(
                 # The `phone_validation` module is installed.
                 partner_sudo._onchange_phone_validation()
 
+        if (
+            'company_name' in address_values
+            and not partner_sudo.is_company
+            and partner_sudo.type == 'contact'
+        ):
+            # If partner is an individual, update existing company's name or remove one
+            company_name = address_values['company_name']
+            parent_company = partner_sudo.parent_id
+            partner_sudo.company_name = False
+
+            if company_name and parent_company and parent_company.name != company_name:
+                parent_company.name = company_name
+            elif not company_name:
+                partner_sudo.parent_id = False
```

Comment:
I think removing the link to the parent is too magical and the customer should request the support to do it (or create a new account).